ykrt/src/compile/jitc_yk/jit_ir/mod.rs

```diff
@@ -1800,6 +1795,16 @@ impl Inst {
                 m.trace_body_end = m.trace_body_end.iter().map(|op| mapper(m, op)).collect();
                 Inst::TraceBodyEnd
             }
+            Inst::TraceHeaderEnd => {
+                // Copy the header label into the body while remapping the operands.
+                m.trace_header_end = m.trace_header_end.iter().map(|op| mapper(m, op)).collect();
```

Comment:
Now you've lost me. Why are we remapping the header? Shouldn't the header stay as it was before we peeled?

Reply:
This is where we discover constants from the analyser. [Well, not here, but `opt/mod.rs` calls this, and finds out constants this way.]

Reply:
Oh, this makes more sense after seeing the following commit. Just to check I got this right: you added this because you later use this function for other things than just peeling off a loop (e.g. to propagate analyser info). When peeling the loop, this function is never called with `TraceHeaderEnd` so we never remap this at that time.

Reply:
Yep!